e2e_tests/tests/experiment/test_core.py

```diff
@@ -295,22 +292,32 @@ def test_end_to_end_adaptive() -> None:
     checkpoint.add_metadata({"testing": "metadata"})
     db_check = d.get_checkpoint(checkpoint.uuid)
     # Make sure the checkpoint metadata is correct and correctly saved to the db.
-    assert checkpoint.metadata == {"testing": "metadata"}
+    # Beginning with 0.18 the system contributes a few items to the dict
+    assert checkpoint.metadata.get("testing") == "metadata"
+    assert checkpoint.metadata.keys() == {"format", "framework", "latest_batch", "testing"}
     assert checkpoint.metadata == db_check.metadata
 
     checkpoint.add_metadata({"some_key": "some_value"})
     db_check = d.get_checkpoint(checkpoint.uuid)
-    assert checkpoint.metadata == {"testing": "metadata", "some_key": "some_value"}
+    assert checkpoint.metadata.items() > {"testing": "metadata", "some_key": "some_value"}.items()
```

Comment:
`>` on sets is a super-set check, as I am sure you know. What I did not know until recently is that `dict.items()` is a set-like data structure if values are hashable, which in this test they are.
From https://docs.python.org/3/library/stdtypes.html#dict-views
> If all values are hashable, so that (key, value) pairs are unique and hashable, then the items view is also set-like.

```
>>> d1 = {'abc': 1, 'de': 2}
>>> d2 = {'aaa': 34, 'de': 2, 'abc': 1}
>>> d1.items() < d2.items()
True
>>> {('de', 2)} < d2.items()
True
>>> {('de', 3)} < d2.items()
False
>>> {('abc', 1), ('aaa', 34)} < d1.items()
False
>>> {('abc', 1), ('aaa', 34)} < d2.items()
True
```

Reply:
oh wow I did not know this either